File: URI/1049.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>

	char string[21];

int main(){
	scanf("%s",string);
	if(strcmp(string,"vertebrado")==0){
		scanf("%s",string);
		if(strcmp(string,"ave")==0){
			scanf("%s",string);
			if(strcmp(string,"carnivoro")==0){
				printf("aguia\n");
			}
			if(strcmp(string,"onivoro")==0){
				printf("pomba\n");
			}
		}
		if(strcmp(string,"mamifero")==0){
			scanf("%s",string);
			if(strcmp(string,"onivoro")==0){
				printf("homem\n");
			}
			if(strcmp(string,"herbivoro")==0){
				printf("vaca\n");
			}
		}
	}
	if(strcmp(string,"invertebrado")==0){
		scanf("%s",string);
		if(strcmp(string,"inseto")==0){
			scanf("%s",string);
			if(strcmp(string,"hematofago")==0){
				printf("pulga\n");
			}
			if(strcmp(string,"herbivoro")==0){
				printf("lagarta\n");
			}
		}
		if(strcmp(string,"anelideo")==0){
			scanf("%s",string);
			if(strcmp(string,"hematofago")==0){
				printf("sanguessuga\n");
			}
			if(strcmp(string,"onivoro")==0){
				printf("minhoca\n");
			}			
		}
	}

	return 0;
}

File: URI/1049.py
def comparar_string(a,b):
	if len(a) != len(b):
		return False
	for i in range(len(a)):
		if a[i] != b[i]:
			return False
	return True

palavra1 = input()
palavra2 = input()
palavra3 = input()

if comparar_string(palavra1,"vertebrado"):
	if comparar_string(palavra2,"ave"):
		if comparar_string(palavra3,"carnivoro"):
			print("aguia")
		elif comparar_string(palavra3,"onivoro"):
			print("pomba")
	elif comparar_string(palavra2,"mamifero"):
		if comparar_string(palavra3,"onivoro"):
			print("homem")
		elif comparar_string(palavra3,"herbivoro"):
			print("vaca") 
elif comparar_string(palavra1,"invertebrado"):
	if comparar_string(palavra2,"inseto"):
		if comparar_string(palavra3,"hematofago"):
			print("pulga")
		elif comparar_string(palavra3,"herbivoro"):
			print("lagarta")
	elif comparar_string(palavra2,"anelideo"):
		if comparar_string(palavra3,"hematofago"):
			print("sanguessuga")
		elif comparar_string(palavra3,"onivoro"):
			print("minhoca")